crates/store/src/server/api.rs

```diff
@@ -593,30 +578,24 @@ impl api_server::Api for StoreApi {
     )]
     async fn list_accounts(
         &self,
-        _request: tonic::Request<ListAccountsRequest>,
+        _request: Request<ListAccountsRequest>,
     ) -> Result<Response<ListAccountsResponse>, Status> {
-        let accounts = self
-            .state
-            .list_accounts()
-            .await
-            .map_err(internal_error)?
-            .iter()
-            .map(Into::into)
-            .collect();
+        let accounts = self.state.list_accounts().await?.iter().map(Into::into).collect();
         Ok(Response::new(ListAccountsResponse { accounts }))
     }
 }
 
 // UTILITIES
 // ================================================================================================
 
-/// Formats an error
-fn internal_error<E: core::fmt::Debug>(err: E) -> Status {
-    Status::internal(format!("{:?}", err))
+/// Formats an "Internal error" error
+fn internal_error<E: core::fmt::Display>(err: E) -> Status {
```

Comment:
Do we still need this one?

Reply:
Yes, it's still used in 5 places. Actually we don't really need this at all, but it is a convenient method which allows us just to write `.map_err(internal_error)` instead of `.map_err(|err| Status::internal(err.to_string()))`.

Reply:
I thought maybe those would be covered by just using `.context(..)` 👍